pages/index.tsx

```diff
@@ -2,25 +2,81 @@ import type { NextPage } from "next";
 import React, { useState, useEffect } from "react";
 import { EuiText, EuiFieldText, EuiFlexGroup, EuiFlexItem, EuiButton } from "@elastic/eui";
 import { useRouter } from 'next/router';
+import Link from 'next/link'
+import { database } from '../firebaseConfig';
+import { setDoc, doc, collection, getDocs, where, query, DocumentData } from "firebase/firestore";
+
+const roomsCollection = collection(database,'rooms');
 
 const Home: NextPage = () => {
   const router = useRouter();
-  const [newRoom, setNewRoom] = useState("");
+  const [newRoom, setNewRoom] = useState('');
   const [joinRoom, setJoinRoom] = useState('');
+  const [roomList, setRoomList] = useState<any[]>([]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const getActiveRooms = () => {
+    let docs: React.SetStateAction<any[]> = [];
+    getDocs(roomsCollection)
+      .then((data) => {
+        data.forEach((doc) => {
+          let id = doc.id;
+          let rd = doc.data();
+          rd.id = id;
+          docs.push(rd);
+        })
+        setRoomList(docs)
+      })
+  };
+
+  const createNewRoom = () => {
+    const currentRoom = newRoom;
+    const docID = "randomID"+ currentRoom
+    setDoc(doc(database, "rooms", docID), {
+      room: currentRoom,
+      users: []
+    }).then(() => {
+      setNewRoom('');
+    }).then(() => {
+      router.push(`/room/${newRoom}?roomid=${docID}`);
+    })
+  }
```

Comment:
This function sends a new document to FB. I'm quirkily creating the doc ID right now. We should set this to a random string in the future. After the room is added to FB, it forwards the user to the room page.